tasks/main.yaml

```diff
@@ -0,0 +1,100 @@
+  tags: [capra_lvm_filesystem_prerequisites]
+
+- vars:
+    _capra_lvm_filesystem_path: "{{ item.path | default(capra_lvm_filesystem_parent_path ~ '/' ~ item.name) }}"
+    _capra_lvm_filesystem_state: "{{ item.state | default('present')}}"
+  block:
+    - name: "Unmount volume"
+      mount:
+        path: "{{ _capra_lvm_filesystem_path }}"
+        state: absent
+      with_list: "{{ capra_lvm_filesystem_specs }}"
+      when: _capra_lvm_filesystem_state == 'absent'
+      tags: [capra_lvm_filesystem_remove]
+
+    - name: "Remove logical volume"
+      lvol:
+        vg: "{{ item.vg | default(capra_lvm_filesystem_vg) }}"
+        lv: "{{ item.name | mandatory }}"
+        size: "{{ item.size | human_to_bytes }}B"
+        opts: "{{ capra_lvm_filesystem_lvcreate_options }}"
+        state: absent
+        force: true
+      with_list: "{{ capra_lvm_filesystem_specs }}"
+      when: _capra_lvm_filesystem_state == 'absent'
+      tags: [capra_lvm_filesystem_remove]
+
+    - name: "Create logical volume"
+      lvol:
+        vg: "{{ item.vg | default(capra_lvm_filesystem_vg) }}"
+        lv: "{{ item.name | mandatory }}"
+        size: "{{ item.size | human_to_bytes }}B"
+        opts: "{{ capra_lvm_filesystem_lvcreate_options }}"
+        state: present
+      with_list: "{{ capra_lvm_filesystem_specs }}"
+      when: _capra_lvm_filesystem_state == 'present'
+      tags: [capra_lvm_filesystem_create]
+
+    - name: "Create filesystem"
+      filesystem:
+        fstype: "{{ item.fstype | default(capra_lvm_filesystem_fstype) }}"
+        dev: "/dev/{{ item.vg | default(capra_lvm_filesystem_vg) }}/{{ item.name }}"
+        opts: "{{ capra_lvm_filesystem_filesystem_options }}"
+        resizefs: true
+      with_list: "{{ capra_lvm_filesystem_specs }}"
+      when: _capra_lvm_filesystem_state == 'present'
+      tags: [capra_lvm_filesystem_create]
+
+    - name: "Lookup volume UUIDs"
+      command: "/sbin/blkid -c /dev/null -o value -s UUID {{ '/dev/' ~ ((item.vg | default(capra_lvm_filesystem_vg)) ~ '/' ~ item.name) | quote }}"
+      changed_when: false
+      register: _capra_lvm_filesystem_blkid
+      with_list: "{{ capra_lvm_filesystem_specs }}"
+      tags: [always]
+      check_mode: false
+      when: _capra_lvm_filesystem_state == 'present'
+
+    - name: "Mount volume"
+      mount:
+        path: "{{ item.1.path | default(capra_lvm_filesystem_parent_path ~ '/' ~ item.1.name) }}"
+        src: "UUID={{ _capra_lvm_filesystem_blkid.results[item.0].stdout }}"
+        fstype: "{{ item.1.fstype | default(capra_lvm_filesystem_fstype) }}"
+        opts: "{{ item.1.options | default(capra_lvm_filesystem_options) }}"
+        state: mounted
+      with_indexed_items: "{{ capra_lvm_filesystem_specs }}"
+      when: (item.1.state | default('present')) == 'present'
+      tags: [capra_lvm_filesystem_create]
+
+    - name: "Set volume permissions"
+      file:
+        path: "{{ _capra_lvm_filesystem_path }}"
+        state: directory
```

Comment:
This should probably not create the directory if it weren't to exist for some reason. That must happen before the mount

Reply:
Got that covered by not touching the files if the mount failed.